hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileScanner.java

```diff
@@ -140,4 +140,10 @@ public interface HFileScanner extends Shipper, Closeable {
    */
   @Override
   void close();
+
+  /**
+   * Returns the block size in bytes for the current block. Will only return a value once per block,
+   * otherwise 0. Used for calculating block IO in ScannerContext.
+   */
+  int getCurrentBlockSizeOnce();
```

Comment:
This API design is a bit strange... Let me take a look on the usage...

Reply:
That seems intuitive and reasonable. Done. Please see latest commit.